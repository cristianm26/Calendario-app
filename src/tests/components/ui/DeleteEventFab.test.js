import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import thunk from "redux-thunk";
import configureStore from 'redux-mock-store';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/events';

jest.fn('../../../actions/events', () => ({
    eventStartDelete: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab/>
    </Provider>
)
describe('Pruebas en deleteEventFab', () => {
    test('debe de mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de llamar el eventStartDelete al hacer click ', () => {
        wrapper.find('button').prop('onClick')();


        expect( eventStartDelete ).toHaveBeenCalled();


    })
    
    
})
