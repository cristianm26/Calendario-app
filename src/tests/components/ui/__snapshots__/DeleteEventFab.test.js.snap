// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en deleteEventFab debe de mostrarse correctamente  1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
    Borrar Evento
  </span>
</button>
`;
